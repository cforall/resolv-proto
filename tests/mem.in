T new
T cons T T
%%
cons( 1 1 )
cons( 1 new() )
cons( cons( 1 new() ) 1 )
cons( cons( 1 new() ) new() )
cons( cons( new() new() ) 1 ) 
new()

