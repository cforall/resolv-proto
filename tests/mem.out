
[1 / (0,1,0)]{T@cons => 1} cons{T}( 1 1 )

[1 / (0,2,0)]{[T@cons, T@new] => 1} cons{T}( 1 new{T}( ) )

[1 / (0,3,0)]{[T@cons, T@cons, T@new] => 1} cons{T}( cons{T}( 1 new{T}( ) ) 1 )

[1 / (0,4,0)]{[T@cons, T@cons, T@new, T@new] => 1} cons{T}( cons{T}( 1 new{T}( ) ) new{T}( ) )

[1 / (0,4,0)]{[T@cons, T@cons, T@new, T@new] => 1} cons{T}( cons{T}( new{T}( ) new{T}( ) ) 1 )

ERROR: unbound type variables in new( ): T@new => ???
